sys/include/net/sock/flags.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2016 Freie Univesität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_sock_flags   Common flags for @ref net_sock
+ * @ingroup     net_sock
+ * @brief       Common flags for usage with @ref net_sock
+ * @{
+ *
+ * @file
+ * @brief   Common flags definitions for usage with @ref net_sock
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef NET_SOCK_FLAGS_H_
+#define NET_SOCK_FLAGS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Common flags for @ref net_conn
+ * @{
+ */
+#define SOCK_FLAGS_REUSE_EP     (0x00000001)    /**< allow to reuse end point on bind */
+/** @} */
```

Comment:
>> Can't we merge this with another header? Maybe directly into sock.h?
> 
> Need to think about it, but probably you are right.

IMO we can drop sock.h. But as the endpoints (for tcp and udp) are
identical, maybe they can move to a shared ep.h?

>> And do we really need space for so many flags?
> 
> Problem with flags is always, that you not just can use `unsigned` (otherwise I would prefer it, since the compiler optimizes it). But true, on non 32-platforms the code size would grow anyway, so maybe a non-32-bit value is better. For now it is the only flag, since that was the only option that came to our mind during the meeting, so uint8_t might suffice (in case other flags come to mind later). Or shall we go for uin16_t just in case?

Why can we not just use unsigned? Isn't that fine until we have up 16 flags?

uint8_t or uint16_t would probably involve extra masking code for 32bit
platforms.